src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -914,6 +914,49 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, config: '' }
+      end
+
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
> The more we test the better.

Sorry but this is not true. It has to be the right balance between test coverage and flexibility to adopt your app. I agree with @evanrolfe, test the error condition, not the text.